bin/specextract

```diff
@@ -828,860 +753,1086 @@ def clip_wmap(wmapfile,threshold,tmpdir):
+    chipy = args["chipy"]
 
-        # Use pycrates to retrieve ccd_id values from user-input event
-        # extraction region. Compute the mode of the ccd_id values
-        # stored in the array.
+    weight = args["weight"]
+    weight_rmf = args["weight_rmf"]
 
-        chipid_vals = pcr.get_colvals(cr, "chip_id")
+    null_rmffile = args["null_rmffile"]
+    rmffile_ccd = args["rmffile_ccd"]
 
-        n_elements = list(range(10))
+    dobpix = args["dobpix"]
+    dobkgresp = args["dobkgresp"]
+    wmap_clip = args["wmap_clip"]
 
-        for i in range(0,10):
-            zeros = numpy.array(chipid_vals)-i
-            n_elements[i] = len(chipid_vals)-len(numpy.nonzero(zeros)[0])
+    tmpdir = args["tmpdir"]
+    clobber = args["clobber"]
+    verbose = args["verbose"]
+    iteminfostr = args["iteminfostr"]
+    pars = args["pars_specextract"]
 
-        mode = numpy.where(n_elements==numpy.max(n_elements))[0]
-        return mode[0]
+    try:
+        bpix = args["bpix"]
+    except KeyError:
+        pass
 
-    elif colname in ["x","y","chipx","chipy"]:
+    tdetwmap = args["tdetwmap"]
+    feffile = args["feffile"]
+    weightfile = None
+    
+    ###########################
+    #
+    # create ACIS RMF
+    #
+    ###########################
 
-        if colname in ["x","y"]:
-            bin_setting="[bin sky=2]"
+    with new_pfiles_environment(ardlib=True,copyuser=False):
+        if dobpix:
+            set_badpix(filename,bpix,instrument,verbose)
+    
+        infostr = f"Creating {srcbkg} RMF {iteminfostr}"
 
-        elif colname in ["chipx","chipy"]:
-            bin_setting="[bin chipx=2,chipy=2]"
+        if null_rmffile:
+            v1(f"WARNING: Setting rmffile parameter (and calquiz calfile) to '{rmffile_ccd}'.\n")
 
-        dmstat(file+bin_setting)
+        rmftool = determine_rmf_tool(phafile,rmffile_ccd,verbose)
 
-        max_cnts_src_pos = dmstat.out_max_loc
+        try:
+            if any([srcbkg == "src", srcbkg == "bkg" and dobkgresp]):
+                if all([weight == "yes",weight_rmf == "yes"]):
+                    respfile = build_rmf_ext(rmftool,ptype,full_outroot,
+                                             ebin,rmfbin,clobber,verbose,
+                                             phafile,weightfile,wmap_clip,
+                                             tdetwmap, infostr)
 
-        src_x = max_cnts_src_pos.split(",")[0]
-        src_y = max_cnts_src_pos.split(",")[1]
+                else:
+                    respfile = build_rmf_ps(rmftool,filename,fullfile,ptype,full_outroot,
+                                            ebin,rmfbin,clobber,verbose,phafile,feffile,
+                                            ccd_id,chipx,chipy,infostr)
 
-        if colname == "x":
-            return src_x
+                try:
+                    add_tool_history(respfile,toolname,pars,toolversion=__revision__)
+                except OSError:
+                    v3("skip writing file history to RMF")
```

Comment:
Is it worth this `try/except` block around `add_tool_history` here? I can't think of a case where it should fail but it's okay to carry on, but maybe there's a helpdesk ticket I'm forgetting.